cmd/agent/app/builder_test.go

```diff
@@ -268,17 +267,19 @@ func TestCreateCollectorProxy(t *testing.T) {
 			require.NoError(t, err)
 			proxy.GetReporter().EmitBatch(context.Background(), jaeger.NewBatch())
 			metricsFactory.AssertCounterMetrics(t, test.metric)
+			defer proxy.Close()
```

Comment:
I think this should go after L263 instead as
```
if err != nil {
    defer proxy.Close()
}
```
Also, it's better to wrap each loop iteration in `t.Run("", func()...)` in this case because otherwise all defers are accumulated till the end of the loop